zap/src/main/java/org/zaproxy/zap/eventBus/SimpleEventBus.java

```diff
@@ -284,11 +255,10 @@ public String getPublisherName() {
 
     private static class RegisteredPublisher {
         private EventPublisher publisher;
-        private String[] eventTypes;
+        private List<String> eventTypes;
```

Comment:
is it possible to make it as Set/LinkedHashSet so that contains complexity can be reduced.

Reply:
We don't need order here so `HashSet` would be enough (though I doubt that this a performance issue, there are not that many events).

Reply:
I thought there are more event types.

Reply:
Well it depends on the publisher some have more than others, but each one does not have many (less than 10).

Reply:
Maybe that's already enough to see a difference, will check...

Reply:
oh.. ok. then makes sense.

Reply:
It comes in as an array in `registerConsumer` anyway and we never modify it. In fact, maybe it should just be made immutable.

Reply:
I just pushed an update which makes these `eventType` lists immutable

Reply:
IMHO, I think in registerConsumer and in registerPublisher we are running contains search and incase there are 10 consumers each with 10 event types then in worst case it will be 100 searches in ``` getConsumer ``` call. As we are already converting array to List we can instead convert to Set if we are not accessing using index.

Reply:
Sure, sounds good to me

Reply:
For the record, with 100 searches it's still a rounding error, you would need way more to see the difference (though a Set is clearly the best choice, good performance in all cases).